libs/sdk-core/src/models.rs

```diff
@@ -637,6 +645,191 @@ pub struct ClosedChannelPaymentDetails {
+        }
+    }
+}
+
+impl FromSql for OpeningFeeParams {
+    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
+        serde_json::from_str(value.as_str()?).map_err(|_| FromSqlError::InvalidType)
+    }
+}
+
+impl ToSql for OpeningFeeParams {
+    fn to_sql(&self) -> rusqlite::Result<ToSqlOutput<'_>> {
+        Ok(ToSqlOutput::from(
+            serde_json::to_string(&self).map_err(|_| FromSqlError::InvalidType)?,
+        ))
+    }
+}
+
+pub enum DynamicFeeType {
+    Cheapest,
+    Longest,
+}
+
+/// See [OpeningFeeParamsMenu::try_from]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct OpeningFeeParamsMenu {
+    pub vals: Vec<OpeningFeeParams>,
+}
+
+impl OpeningFeeParamsMenu {
+    /// Initializes and validates itself.
+    ///
+    /// This struct should not be persisted as such, because validation happens dynamically based on
+    /// the current time. At a later point in time, any previously-validated [OpeningFeeParamsMenu]
+    /// could be invalid. Therefore, the [OpeningFeeParamsMenu] should always be initialized on-the-fly.
+    pub fn try_from(vals: Vec<grpc::OpeningFeeParams>) -> Result<Self> {
+        let temp = Self {
+            vals: vals
+                .into_iter()
+                .map(|ofp| ofp.into())
+                .collect::<Vec<OpeningFeeParams>>(),
+        };
+        temp.validate().map(|_| temp)
+    }
+
+    fn validate(&self) -> Result<()> {
+        // opening_fee_params_menu fees must be in ascending order
+        let is_ordered = self.vals.windows(2).all(|ofp| {
+            let larger_min_msat_fee = ofp[0].min_msat < ofp[1].min_msat;
+            let equal_min_msat_fee = ofp[0].min_msat == ofp[1].min_msat;
+
+            let larger_proportional = ofp[0].proportional < ofp[1].proportional;
+            let equal_proportional = ofp[0].proportional == ofp[1].proportional;
+
+            (larger_min_msat_fee && equal_proportional)
+                || (equal_min_msat_fee && larger_proportional)
+                || (larger_min_msat_fee && larger_proportional)
+        });
+        ensure!(is_ordered, "Validation failed: fee params are not ordered");
+
+        // Menu must not contain any item with `valid_until` in the past
+        let is_expired = self.vals.iter().any(|ofp| match ofp.valid_until_date() {
+            Ok(valid_until) => Utc::now() > valid_until,
+            Err(_) => {
+                warn!("Failed to parse valid_until for OpeningFeeParams: {ofp:?}");
+                false
+            }
+        });
+        ensure!(!is_expired, "Validation failed: expired fee params found");
+
+        Ok(())
+    }
+
+    pub(crate) fn get_cheapest_opening_fee_params(&self) -> Result<OpeningFeeParams> {
+        self.vals.first().cloned().ok_or(anyhow!(
+            "The LSP doesn't support opening new channels: Dynamic fees menu contains no values"
+        ))
+    }
+
+    pub(crate) fn get_longest_valid_opening_fee_params(&self) -> Result<OpeningFeeParams> {
```

Comment:
Let's rename this to `get_48h_opening_fee_params` to reflect what it is actually doing.

Reply:
Done